src/eth_provider/database/types/transaction.rs

```diff
@@ -15,6 +19,57 @@ impl From<StoredTransaction> for Transaction {
     }
 }
 
+#[cfg(any(test, feature = "arbitrary"))]
+impl<'a> arbitrary::Arbitrary<'a> for StoredTransaction {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let transaction = TransactionSigned::arbitrary(u)?;
+
+        let mut access_list: Option<Vec<reth_rpc_types::AccessListItem>> = Default::default();
+
+        if let Some(list) = transaction.access_list() {
+            access_list = Some(Default::default());
+            for item in &list.0 {
+                let address = item.address;
+                let storage_keys = &item.storage_keys;
+                access_list
+                    .as_mut()
+                    .unwrap()
+                    .push(reth_rpc_types::AccessListItem { address, storage_keys: storage_keys.to_vec() });
+            }
+        }
```

Comment:
I find this part a little strange, can't we use a `Vec<reth_rpc_types::AccessListItem>` and push the items in it? Or even just an `into_iter().map()` on `transaction.access_list()`?

Reply:
Yes you are right, just simplified this part.